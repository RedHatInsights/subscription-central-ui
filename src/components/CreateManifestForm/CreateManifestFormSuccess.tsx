import React, { FC } from 'react';
import {
  Button,
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStateVariant,
  Title
} from '@patternfly/react-core';
import CheckCircleIcon from '@patternfly/react-icons/dist/js/icons/check-circle-icon';

interface CreateManifestFormSuccessProps {
  manifestName: string;
  handleModalToggle: any;
}

const CreateManifestFormSuccess: FC<CreateManifestFormSuccessProps> = ({
  manifestName,
  handleModalToggle
}) => {
  return (
    <EmptyState variant={EmptyStateVariant.small}>
      <EmptyStateIcon icon={CheckCircleIcon} color="var(--pf-global--success-color--100)" />
      <Title headingLevel="h3">Success</Title>
      <EmptyStateBody>
        <h4>
          Your new manifest, <strong>{manifestName}</strong>, has been successfully created.
        </h4>
        <Button
          key="close-window"
          variant="primary"
          onClick={handleModalToggle}
          style={{ marginTop: '20px' }}
        >
          Return to page
        </Button>
      </EmptyStateBody>
    </EmptyState>
  );
};

export default CreateManifestFormSuccess;
