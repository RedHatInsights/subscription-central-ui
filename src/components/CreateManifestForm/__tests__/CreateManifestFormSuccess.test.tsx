import React from 'react';
import { render } from '@testing-library/react';
import CreateManifestFormSuccess from '../CreateManifestFormSuccess';

describe('Create Manifest Form Success Message', () => {
  it('renders correctly', () => {
    const props = {
      manifestName: 'new-manifest',
      handleModalToggle: () => 'foo'
    };
    const { container } = render(<CreateManifestFormSuccess {...props} />);
    expect(container).toMatchSnapshot();
  });
});
