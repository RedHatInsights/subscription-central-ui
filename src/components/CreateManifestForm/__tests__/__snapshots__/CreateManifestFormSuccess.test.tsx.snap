// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Manifest Form Success Message renders correctly 1`] = `
<div>
  <div
    class="pf-c-empty-state pf-m-sm"
  >
    <div
      class="pf-c-empty-state__content"
    >
      <svg
        aria-hidden="true"
        class="pf-c-empty-state__icon"
        fill="var(--pf-global--success-color--100)"
        height="1em"
        role="img"
        style="vertical-align: -0.125em;"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
        />
      </svg>
      <h3
        class="pf-c-title pf-m-lg"
      >
        Success
      </h3>
      <div
        class="pf-c-empty-state__body"
      >
        <h4>
          Your new manifest, 
          <strong>
            new-manifest
          </strong>
          , has been successfully created.
        </h4>
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-primary"
          data-ouia-component-id="OUIA-Generated-Button-primary-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          style="margin-top: 20px;"
          type="button"
        >
          Return to page
        </button>
      </div>
    </div>
  </div>
</div>
`;
